File: src/enums/production.py
from enum import Enum
from typing import List


class Production(Enum):
    RUSSIAN = "russian"
    KOREAN = "korean"
    TURKISH = "turkish"
    FOREIGN = "foreign"

    def to_rus(self) -> str:
        production2rus = {
            Production.RUSSIAN: "российское",
            Production.KOREAN: "корейское",
            Production.TURKISH: "турецкое",
            Production.FOREIGN: "зарубежное"
        }

        return production2rus[self]

    @classmethod
    def from_countries(cls: "Production", countries: List[str]) -> "List[Production]":
        country2production = {
            "Россия": Production.RUSSIAN,
            "СССР": Production.RUSSIAN,
            "Корея Южная": Production.KOREAN,
            "Турция": Production.TURKISH
        }

        return list({country2production.get(country, Production.FOREIGN) for country in countries})

File: web/js/entities/production.js
function Production(value) {
    this.value = value
    this.options = {
        "russian": "российское",
        "korean": "корейское",
        "turkish": "турецкое",
        "foreign": "зарубежное"
    }
}

Production.prototype.ToRus = function() {
    return this.options[this.value]
}

function ProductionList(productions) {
    this.productions = productions.map(production => new Production(production))
}

ProductionList.prototype.ToRus = function() {
    return this.productions.map(production => production.ToRus()).join(", ")
}
